File: Graduway.Client/graduway-app/src/app/Models/task.ts
export interface ITask {
  id: number;
  title: string;
  description: string;
  estimate: number;
  employeeId: number;
  priorityLevelId: number;
  stateId: number;
  priorityLevelLabel: string;
  stateLabel: string;
}
class Task {
  id?: number;
  title: string;
  description: string;
  employeeId: number;
  priorityLevelId: number;
  stateId: number;
  estimate: number;
  priorityLevelLabel: string;
  stateLabel: string;

  constructor(data?: Partial<ITask>) {
    this.id = data?.id;
    this.title = data?.title;
    this.description = data?.description;
    this.employeeId = data?.employeeId;
    this.priorityLevelId = data?.priorityLevelId;
    this.stateId = data?.stateId;
    this.estimate = data?.estimate;
    this.priorityLevelLabel = data?.priorityLevelLabel;
    this.stateLabel = data?.stateLabel;
  }
}

export default Task;

File: Graduway.API/Graduway.Web/Graduway.DAL/Entities/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Graduway.DAL.Entities
{
    public class Task
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int? Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public double Estimate { get; set; }

        public int StateId { get; set; }
        public int PriorityLevelId { get; set; }
        public int EmployeeId { get; set; }
        public State State { get; set; }
        public PriorityLevel PriorityLevel { get; set; }
        public Employee Employee { get; set; }
    }
}
